python/tests/stress/arcticdb/version_store/test_mem_leaks.py

```diff
@@ -182,16 +310,61 @@ def proc_to_examine():
          run the test from command line again to assure it runs ok before commit 
 
     """
-    max_mem_bytes = 450_623_040
+    max_mem_bytes = 295_623_040
+
+    check_process_memory_leaks(proc_to_examine, 20, max_mem_bytes, 80.0)
+
+@pytest.mark.skipif(WINDOWS, reason="Not enough storage on Windows runners")
+@pytest.mark.skipif(MACOS, reason="Problem on MacOs")
+@SKIP_CONDA_MARK # Conda CI runner doesn't have enough storage to perform these stress tests
+def test_mem_leak_querybuilder_standard(arctic_library_lmdb):
+    """
+        This test uses old approach with iterations.
+        It is created for comparison with the new approach
+        with memray
+        (If memray is good in future we could drop the old approach)
+    """
+    lib: Library = arctic_library_lmdb
+
+    df = construct_df_querybuilder_tests(size=2000000)
+    size = df.shape[0]
+    start_date = df.iloc[0].name
+    end_date = df.iloc[0].name
+
+    symbol = "test"
+    lib.write(symbol, df)
+    del df
+    gc.collect()
+
+    queries = [query_group(), query_group_bigger_result(), query_apply(random_string(10)), query_resample()]
+
+    def proc_to_examine():
+        queries = [query_group(), 
+                    query_group_bigger_result(), 
+                    query_apply(random_string(10)), 
+                    query_resample(),
+                    query_row_range(size),
+                    query_date_range(start_date, end_date)]
+        for q in queries:
+            data: pd.DataFrame = lib.read(symbol, query_builder=q).data
+            print_info(data, q)
+            del data
+            gc.collect()
+
+    max_mem_bytes = 550_623_040
+
+    check_process_memory_leaks(proc_to_examine, 10, max_mem_bytes, 80.0)
+
+    del lib, queries
+    gc.collect()
+    time.sleep(10)
```

Comment:
See how psutils sees memory without sleep and with sleep. Ovviously there are processes that require 'calm' process in order for external tools on which psutility is based to measure properly memory.

Thus having NO sleep it is rather road to flaky tests - I have observed that already.

-----------------------------------------------------------------------------
  WITHOUT WAITS 10 SEC
-----------------------------------------------------------------------------

Process initial RSS  1700.16MB/[1782743040] 
Starting watched code ...........
Lets pause for 0 secs so GC to kick in
Iter No[0]    : Process added (or if number is negative - process released)  -36.97MB/[-38768640] .  Avail memory:  43992.38MB/[46129352704]  Used Mem: 8.7%
  Overall stats : Process growth since start  -36.97MB/[-38768640]  AVG growth per iter  -36.97MB/[-38768640.0] 
  Minimum growth so far:  -36.97MB/[-38768640] 
  Maximum growth so far:  -36.97MB/[-38768640] 
  Number of times there was 50% drop in memory: 0
Starting watched code ...........
Lets pause for 0 secs so GC to kick in
Iter No[1]    : Process added (or if number is negative - process released)  0.17MB/[180224] .  Avail memory:  44074.61MB/[46215573504]  Used Mem: 8.5%
  Overall stats : Process growth since start  -36.80MB/[-38588416]  AVG growth per iter  -18.40MB/[-19294208.0] 
  Minimum growth so far:  -36.97MB/[-38768640] 
  Maximum growth so far:  -36.80MB/[-38588416] 
  Number of times there was 50% drop in memory: 1
Starting watched code ...........
Lets pause for 0 secs so GC to kick in
Iter No[2]    : Process added (or if number is negative - process released)  13.77MB/[14442496] .  Avail memory:  44088.59MB/[46230241280]  Used Mem: 8.5%
  Overall stats : Process growth since start  -23.03MB/[-24145920]  AVG growth per iter  -7.68MB/[-8048640.0] 
  Minimum growth so far:  -36.97MB/[-38768640] 
  Maximum growth so far:  -23.03MB/[-24145920] 
  Number of times there was 50% drop in memory: 2
Starting watched code ...........
Lets pause for 0 secs so GC to kick in
Iter No[3]    : Process added (or if number is negative - process released)  3.79MB/[3977216] .  Avail memory:  44066.72MB/[46207307776]  Used Mem: 8.5%
  Overall stats : Process growth since start  -19.23MB/[-20168704]  AVG growth per iter  -4.81MB/[-5042176.0] 
  Minimum growth so far:  -36.97MB/[-38768640] 
  Maximum growth so far:  -19.23MB/[-20168704] 
  Number of times there was 50% drop in memory: 3
Starting watched code ...........
Lets pause for 0 secs so GC to kick in
Iter No[4]    : Process added (or if number is negative - process released)  43.75MB/[45879296] .  Avail memory:  44020.70MB/[46159044608]  Used Mem: 8.6%
  Overall stats : Process growth since start  24.52MB/[25710592]  AVG growth per iter  4.90MB/[5142118.4] 
  Minimum growth so far:  -36.97MB/[-38768640] 
  Maximum growth so far:  24.52MB/[25710592] 
  Number of times there was 50% drop in memory: 3
Starting watched code ...........
Lets pause for 0 secs so GC to kick in
Iter No[5]    : Process added (or if number is negative - process released)  103.92MB/[108969984] .  Avail memory:  43923.30MB/[46056919040]  Used Mem: 8.8%
  Overall stats : Process growth since start  128.44MB/[134680576]  AVG growth per iter  21.41MB/[22446762.666666668] 
  Minimum growth so far:  -36.97MB/[-38768640] 
  Maximum growth so far:  128.44MB/[134680576] 
  Number of times there was 50% drop in memory: 3
Starting watched code ...........
Lets pause for 0 secs so GC to kick in
Iter No[6]    : Process added (or if number is negative - process released)  -22.30MB/[-23379968] .  Avail memory:  43941.50MB/[46076002304]  Used Mem: 8.8%
  Overall stats : Process growth since start  106.14MB/[111300608]  AVG growth per iter  15.16MB/[15900086.857142856] 
  Minimum growth so far:  -36.97MB/[-38768640] 
  Maximum growth so far:  128.44MB/[134680576] 
  Number of times there was 50% drop in memory: 3
Starting watched code ...........
Lets pause for 0 secs so GC to kick in
Iter No[7]    : Process added (or if number is negative - process released)  -138.72MB/[-145457152] .  Avail memory:  44065.91MB/[46206459904]  Used Mem: 8.5%
  Overall stats : Process growth since start  -32.57MB/[-34156544]  AVG growth per iter  -4.07MB/[-4269568.0] 
  Minimum growth so far:  -36.97MB/[-38768640] 
  Maximum growth so far:  128.44MB/[134680576] 
  Number of times there was 50% drop in memory: 4
  

-----------------------------------------------------------------------------
  WITH WAITS 10 SEC
-----------------------------------------------------------------------------
Process initial RSS  1734.86MB/[1819131904] 
Starting watched code ...........
Lets pause for 10 secs so GC to kick in
Iter No[0]    : Process added (or if number is negative - process released)  45.19MB/[47386624] .  Avail memory:  43974.38MB/[46110478336]  Used Mem: 8.7%
  Overall stats : Process growth since start  45.19MB/[47386624]  AVG growth per iter  45.19MB/[47386624.0] 
  Minimum growth so far:  45.19MB/[47386624] 
  Maximum growth so far:  45.19MB/[47386624] 
  Number of times there was 50% drop in memory: 0
Starting watched code ...........
Lets pause for 10 secs so GC to kick in
Iter No[1]    : Process added (or if number is negative - process released)  -7.00MB/[-7344128] .  Avail memory:  43968.44MB/[46104248320]  Used Mem: 8.7%
  Overall stats : Process growth since start  38.19MB/[40042496]  AVG growth per iter  19.09MB/[20021248.0] 
  Minimum growth so far:  38.19MB/[40042496] 
  Maximum growth so far:  45.19MB/[47386624] 
  Number of times there was 50% drop in memory: 0
Starting watched code ...........
Lets pause for 10 secs so GC to kick in
Iter No[2]    : Process added (or if number is negative - process released)  -82.82MB/[-86847488] .  Avail memory:  44054.60MB/[46194597888]  Used Mem: 8.6%
  Overall stats : Process growth since start  -44.64MB/[-46804992]  AVG growth per iter  -14.88MB/[-15601664.0] 
  Minimum growth so far:  -44.64MB/[-46804992] 
  Maximum growth so far:  45.19MB/[47386624] 
  Number of times there was 50% drop in memory: 1
Starting watched code ...........
Lets pause for 10 secs so GC to kick in
Iter No[3]    : Process added (or if number is negative - process released)  88.26MB/[92545024] .  Avail memory:  43922.98MB/[46056579072]  Used Mem: 8.8%
  Overall stats : Process growth since start  43.62MB/[45740032]  AVG growth per iter  10.91MB/[11435008.0] 
  Minimum growth so far:  -44.64MB/[-46804992] 
  Maximum growth so far:  45.19MB/[47386624] 
  Number of times there was 50% drop in memory: 1
Starting watched code ...........
Lets pause for 10 secs so GC to kick in
Iter No[4]    : Process added (or if number is negative - process released)  -82.00MB/[-85979136] .  Avail memory:  44019.06MB/[46157332480]  Used Mem: 8.6%
  Overall stats : Process growth since start  -38.38MB/[-40239104]  AVG growth per iter  -7.67MB/[-8047820.8] 
  Minimum growth so far:  -44.64MB/[-46804992] 
  Maximum growth so far:  45.19MB/[47386624] 
  Number of times there was 50% drop in memory: 2
Starting watched code ...........
Lets pause for 10 secs so GC to kick in
Iter No[5]    : Process added (or if number is negative - process released)  42.20MB/[44249088] .  Avail memory:  43991.38MB/[46128308224]  Used Mem: 8.7%
  Overall stats : Process growth since start  3.82MB/[4009984]  AVG growth per iter  0.64MB/[668330.6666666666] 
  Minimum growth so far:  -44.64MB/[-46804992] 
  Maximum growth so far:  45.19MB/[47386624] 
  Number of times there was 50% drop in memory: 2
Starting watched code ...........
Lets pause for 10 secs so GC to kick in
Iter No[6]    : Process added (or if number is negative - process released)  -58.56MB/[-61403136] .  Avail memory:  44048.85MB/[46188564480]  Used Mem: 8.6%
  Overall stats : Process growth since start  -54.73MB/[-57393152]  AVG growth per iter  -7.82MB/[-8199021.714285715] 
  Minimum growth so far:  -54.73MB/[-57393152] 
  Maximum growth so far:  45.19MB/[47386624] 
  Number of times there was 50% drop in memory: 3
Starting watched code ...........
Lets pause for 10 secs so GC to kick in